modules/ting/ting.field.inc

```diff
@@ -447,24 +447,57 @@ function ting_field_formatter_view($entity_type, $entity, $field, $instance, $la
         break;
 
       case 'ting_author_default':
-        $creators = array();
-        foreach ($entity->creators as $i => $creator) {
-          // TODO BBS-SAL: Use a variable for holding the search query the same
-          // way it is done with ting_search_register_serie_title.
-          $creators[] = l($creator, 'search/ting/phrase.creator="' . $creator . '"', array('attributes' => array('class' => array('author'))));
-        }
+        /** @var \TingEntity $entity */
+        // Primo uses facets that filters on "last name, first name" where
+        // OpenSearch uses "first name, last name".
+        $creators = array_map(function($creator, $inverted_creator) {
+          return ['default' => $creator, 'inverted' => $inverted_creator];
+        }, $entity->getCreators(), $entity->getInvertedCreators());
+
+        // Array mapped with replacement string as key and the author name as
+        // the value.
+        $replace_array = array_map(function($creator) {
+          return [
+            '@author' => $creator['default'],
+            '@author_inverted' => $creator['inverted'],
+          ];
+        }, $creators);
+
+        // The variable specify whether to use to the default or inverted
+        // form.
+        $search_term = strtr(
+          variable_get('ting_search_register_author', 'phrase.creator="@author"'),
+          $replace_array
+        );
+
+        // The links with the correct author name depending on the whether
+        // @author or @author_inverted is used.
+        $creator_links = array_map(function($creator, $search_term) {
+          return [
+            l($creator['default'],'search/ting/' . $search_term, [
+              'attributes' => [
+                'class' => ['author'],
+              ],
+            ]),
+          ];
+        }, $creators, $search_term);
```

Comment:
Subsequent arguments to [`array_map`](http://php.net/manual/en/function.array-map.php) should also be arrays of the same length. Here is it a single string.